src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

```diff
@@ -828,13 +828,9 @@ function discoverController(
     if (newSavedQueryId) {
       setAppState({ savedQuery: newSavedQueryId });
     } else {
-      //reset filters and query string, remove savedQuery from state
+      // remove savedQueryId from state
       const state = {
         ...appStateContainer.getState(),
```

Comment:
This what was causing a "loop". Since as a hot fix I just moved `localStorage.get('kibana.userQueryLanguage')` into search_bar logic, this addition logic is not needed anymore. 
query and filters are anyway synced outside of this